pkg/ruler/ruler.go

```diff
@@ -61,14 +61,14 @@ type Config struct {
 
 	EnableSharding   bool // Enable sharding rule groups
 	SearchPendingFor time.Duration
-	LifecyclerConfig ring.LifecyclerConfig
+	RulerRing        RingConfig
```

Comment:
The resulting yaml field name is `rulerring`, which looks a bit repetitive considering we're already inside the `ruler: {}` config. What's your take adding a `yaml:"ring"`? In case you decide to keep the current name, then `ruler_ring` may be better (easier to read for the user).